proposals/static-local-functions.md

```diff
@@ -0,0 +1,41 @@
+# Static local functions
+
+## Summary
+
+Support local functions that disallow capturing state from the enclosing scope.
+
+## Motivation
+
+Avoid unintentionally capturing state from the enclosing context.
+Allow local functions to be used in scenarios where a `static` method is required.
+
+## Detailed design
+
+A local function declared `static` cannot capture state from the enclosing scope.
+As a result, locals, parameters, and `this` from the enclosing scope are not available within a `static` local function.
+
+A `static` local function cannot reference instance members from an implicit or explicit `this` or `base` reference.
+
+A `static` local function may reference `static` members from the enclosing scope.
+
+A `static` local function may reference `constant` definitions from the enclosing scope.
+
+`nameof()` in a `static` local function cannot reference state from the enclosing scope.
```

Comment:
that seems super strange.  why?  nameof is just a constant, so it should be fine, right?